python/vyos/ifconfig/interface.py

```diff
@@ -637,39 +635,51 @@ def add_addr(self, addr):
         >>> j.get_addr()
         ['192.0.2.1/24', '2001:db8::ffff/64']
         """
-
-        # cache new IP address which is assigned to interface
-        self._addr.append(addr)
-
-        # we can not have both DHCP and static IPv4 addresses assigned to an interface
-        if 'dhcp' in self._addr:
-            for addr in self._addr:
-                # do not change below 'if' ordering esle you will get an exception as:
-                #   ValueError: 'dhcp' does not appear to be an IPv4 or IPv6 address
-                if addr != 'dhcp' and is_ipv4(addr):
-                    raise ConfigError(
-                        "Can't configure both static IPv4 and DHCP address on the same interface")
-
-        if addr == 'dhcp':
-            self.dhcp.v4.set()
-        elif addr == 'dhcpv6':
-            self.dhcp.v6.set()
+        # XXX: normalize/compress with ipaddress if calling functions don't?
+        # is subnet mask always passed, and in the same way?
+        ret = True
+
+        # we can't have both DHCP and static IPv4 addresses assigned
+        for a in self._addr:
+            if ( ( addr == 'dhcp' and a != 'dhcpv6' and is_ipv4(a) ) or
+                    ( a == 'dhcp' and addr != 'dhcpv6' and is_ipv4(addr) ) ):
+                raise ConfigError((
```

Comment:
why is it not possible?

Reply:
I don't know, that's what the function did previously. If there's no other limitation, this can be removed?

Reply:
I tried it quickly and an interface can have dhcp and static, but even if this can be removed it should be a separate PR